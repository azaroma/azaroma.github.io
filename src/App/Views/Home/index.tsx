import * as React from "react";
import { H1 } from "../../../components/Headings";

const Welcome: React.SFC = () => (
  <section className="pa2">
    <H1>Welcome</H1>
  </section>
);

export default Welcome;
