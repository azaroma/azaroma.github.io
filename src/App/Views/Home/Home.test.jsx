import { shallow } from "enzyme";
import * as React from "react";
import Home from ".";

describe("Home view", () => {
  const wrapper = shallow(<Home />);
  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toHaveLength(1);
  });

  it("looks as expected", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
