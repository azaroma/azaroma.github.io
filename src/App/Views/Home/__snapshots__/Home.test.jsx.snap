// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view looks as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="f2 f1-ns mv2"
      >
        Welcome
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Welcome",
        "className": "f2 f1-ns mv2",
      },
      "ref": null,
      "rendered": "Welcome",
      "type": "h1",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="f2 f1-ns mv2"
        >
          Welcome
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome",
          "className": "f2 f1-ns mv2",
        },
        "ref": null,
        "rendered": "Welcome",
        "type": "h1",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
